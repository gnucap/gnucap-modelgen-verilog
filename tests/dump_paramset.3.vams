
`include "sub/disciplines.few"
`pragma modelgen expand-paramset
`pragma modelgen nodump-module
`pragma modelgen dump-paramset

module resistor(a,b)
	electrical a, b;

	parameter real r = 1.;

	analog begin
		I(a,b) <+ V(a,b)/r;
	end
endmodule

paramset my_res resistor;
	parameter real r = 1.;
	.r = r;
endparamset;

paramset my_res resistor;
	parameter real R = 1k from [1k:1k];
	.r = R;
endparamset;

paramset my_res resistor;
	parameter real R0 = 1.;
	parameter real R1 = 1.;
	.r = R0 + R1;
endparamset;
