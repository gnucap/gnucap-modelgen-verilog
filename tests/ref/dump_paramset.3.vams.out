// paramsets
module my_res(a, b);
  electrical a, b;
  parameter real r = 1.;
  localparam real __r = r;

  analog begin
    I(a, b) <+ (V(a, b) / __r);
  end

endmodule
module my_res(a, b);
  electrical a, b;
  parameter real R = 1k from [(1k):(1k)];
  localparam real __r = R;

  analog begin
    I(a, b) <+ (V(a, b) / __r);
  end

endmodule
module my_res(a, b);
  electrical a, b;
  parameter real R0 = 1.;
  parameter real R1 = 1.;
  localparam real __r = (R0 + R1);

  analog begin
    I(a, b) <+ (V(a, b) / __r);
  end

endmodule

